### DISCUSSION TITLE
API 4: Plugin manifest changes

### DISCUSSION THREAD

**Discussion by goaaats** [goatcorp/Dalamud#457] [Announcements]
In preparation for the .NET 5 release, we've redesigned the plugin installer and added 4 properties that you will/can add to your plugin manifest.

* `ImageUrls`: Array of links to screenshots/explainer/whatever that will be displayed, 730x380 resolution, maximum amount of images is 5
* `IconUrl`: Link to a square icon for your plugin, needs to be 512x512
* `Punchline`: One-sentence description of your plugin
* `Changelog`: Small description of recent changes to your plugin, will only be shown for people that have the plugin installed

For images: Please use PNG; transparency is OK

If you haven't done so yet, you can also fill out the `Tags` property for your plugin, which should be an array of strings that can aid people in finding your plugin via search.

**If you publish your plugin on the main DalamudPlugins repo:**
You have to upload your icon/images into an ``images`` subfolder for your plugin the official repository.
Icons should be called ``icon.png``, images should be called ``imageX.png`` where X is a number from 1 to 5.
[See an example here.](https://github.com/goatcorp/DalamudPlugins/tree/api4/plugins/owofy/images)

Please make sure that you have the rights to whatever assets you use.

![image](https://user-images.githubusercontent.com/16760685/128862046-984666d7-51bb-4930-a963-7ed826215661.png)

**Comment by PhilippeBakhtiari** [comment]
Trying to use IconUrl gets me this error as the InstallationPath of my game seems to be added to the URL :
2021-10-26 20:32:55.088 +02:00 [ERR] Failed to load image from https://github.com/goatcorp/DalamudPlugins/blob/api4/plugins/owofy/images/icon.png
System.IO.IOException: La syntaxe du nom de fichier, de répertoire ou de volume est incorrecte. : 'E:\SquareE\SquareEnix\FINAL FANTASY XIV - A Realm Reborn\game\https:\github.com\goatcorp\DalamudPlugins\blob\api4\plugins\owofy\images\icon.png'
   at System.IO.FileStream.ValidateFileHandle(SafeFileHandle fileHandle)

**Reply by rlynch-ironnet** [reply]
devPlugin or installed from the in-game window?

**Reply by PhilippeBakhtiari** [reply]
Tried 2 methods : 
- using the inplace IconTester 
- specifying an URI in IconUrl in my own project
Both get the same output.

**Comment by daemitus** [comment]
The iconTester pane uses filepaths from disk and best I can tell, doesn't
do any file manipulation.
The json manifest files use URLs.

Can you paste the manifest you were using so I can take a quick look?

**Reply by PhilippeBakhtiari** [reply]
Noted for the iconTester :) 
Replied below with the manifest.

**Comment by rlynch-ironnet** [comment]
use https://raw.githubusercontent.com/PhilippeBakhtiari/plgimages/main/phoenixgold2.png as your icon url

**Reply by rlynch-ironnet** [reply]
its failing over to a file icon because thats not a URL to an actual image

**Reply by PhilippeBakhtiari** [reply]
Oh my. That makes sense. Trying it !

**Reply by PhilippeBakhtiari** [reply]
It works perfectly ! Thank you very much 👍
